catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/DumpCommand.java

```diff
@@ -312,6 +279,54 @@ protected Object executeWithSubject() throws Exception {
     return null;
   }
 
+  private void handleResult(
+      SourceResponse response,
+      ExecutorService executorService,
+      File dumpDir,
+      AtomicLong resultCount) {
+    if (StringUtils.isNotBlank(zipFileName)) {
+      try {
+        Optional<QueryResponseTransformer> zipCompressionTransformer = getZipCompression();
+        if (zipCompressionTransformer.isPresent()) {
+          BinaryContent binaryContent =
+              zipCompressionTransformer.get().transform(response, zipArgs);
+          if (binaryContent != null) {
+            IOUtils.closeQuietly(binaryContent.getInputStream());
+          }
+          Long resultSize = (long) response.getResults().size();
+          printStatus(resultCount.addAndGet(resultSize));
+        }
+      } catch (InvalidSyntaxException e) {
+        LOGGER.info("No Zip Transformer found.  Unable export metacards to a zip file.");
+      } catch (CatalogTransformerException e) {
+        LOGGER.info("zipCompression transform failed");
+      }
+    } else if (multithreaded > 1) {
+      final List<Result> results = new ArrayList<>(response.getResults());
+      executorService.submit(
```

Comment:
✏️ Time to make this code a private method and use a method reference instead. It will make it more readable for one and since the code is identical to the one in the else, you might be able to refactor to re-use the same code.